added support for haproxy group per service port

ajays20078 opened the pull request:
Added support for haproxy group per service port. Currently both service port had to be part of same hagroup group. Now similar to Vhost, we can have different haproxy group per service port. This allows us to have external and internal services running on same box on different service ports, as they can be part of different haproxy group. This supports,

HAPROXY_0_GROUP
HAPROXY_1_GROUP 
and maps them to corresponding service ports.

It is backward compatible as well, so it supports HAPROXY_GROUP.

If both HAPROXY_0_GROUP and HAPROXY_GROUP are specified, the more specific definition takes preference and the service group will be added to HAPROXY_0_GROUP and not HAPROXY_GROUP.

If there are 2 services HAPROXY_0_PORT and HAPROXY_1_PORT and if HAPROXY_0_GROUP and HAPROXY_GROUP labels are defined. HAPROXY_0_PORT service is added to HAPROXY_0_GROUP and HAPROXY_1_PORT is added to HAPROXY_GROUP since HAPROXY_1_GROUP is not specified for it.

lloesche commented:
Thank you for the PR!

Can you please add a test that tests the correct config generation with two services in two different groups? One using your newly implemented HAPROXY_0_GROUP and one falling back to HAPROXY_GROUP.

ajays20078 commented:
Added test cases for

1. Fallback to HAPROXY_GROUP label for 2 services(on same hosts) when HAPROXY_{n}_GROUP is not specified for any of the services.
2. Using new HAPROXY_{n}_GROUP label with 2 different services(on same hosts), with each service being associated with a different haproxy_group.
3. Hybrid Config where HAPROXY_0_GROUP is specified for first service and HAPROXY_GROUP is also specified without specifying HAPROXY_1_GROUP for second service.

Added documentation as well.

ajays20078 commented:
Is anything else required here?

lloesche commented:
@ajays20078 thank you! Can you rebase on current master and squash into a single commit please? I'll merge it then.

ajays20078 commented:
Done

lloesche commented:
@ajays20078 Github is still showing me `This branch has conflicts that must be resolved`, also there's five commits in this PR now instead of one. If your local branch is clean try something along these lines: `git reset --soft HEAD~5 && git commit -am 'added support for haproxy group per service port' && git push --force`.

ajays20078 commented:
That dint work ,as my local branch was not clean, have forked off  a fresh branch again and committed into that. Opening a different PR for that and closing this. Sorry for too much of confusion.
